app/service/FileService.scala

```diff
@@ -40,48 +40,27 @@ class FileService @Inject()(connector: BindingTariffFilestoreConnector) {
     connector.get(file).map(toFileAttachment(file.size))
   }
 
-  /*
-   * Publishes the file IF it has been successfully scanned.
-   **/
-  def publish(file: FileAttachment)(implicit hc: HeaderCarrier): Future[SubmittedFileAttachment] = {
-
-    import ScanStatus._
-
-    connector.get(file).flatMap {
-
-      case r if r.scanStatus.contains(READY) =>
-        // If the file has been scanned and marked as safe
-        connector.publish(file).map(toPublishedAttachment(file.size))
-
-      case r if r.scanStatus.contains(FAILED) =>
-        // If the file has been quarantined by the virus scanner
-        Logger.warn("File could not be published as it was [Quarantined]. It will be lost.")
-        successful(UnpublishedFileAttachment(r.id, r.fileName, r.mimeType, file.size, "Quarantined"))
-
-      case r =>
-        // If the file has not been scanned yet.
-        Logger.warn("File could not be published as it was [Un-scanned]. It will be lost.")
-        successful(UnpublishedFileAttachment(r.id, r.fileName, r.mimeType, file.size, "Unscanned"))
-    }
-
+  def publish(file: FileAttachment)(implicit hc: HeaderCarrier): Future[PublishedFileAttachment] = {
+    connector.publish(file).map(toPublishedAttachment(file.size))
   }
 
-  def publish(files: Seq[FileAttachment])(implicit headerCarrier: HeaderCarrier): Future[Seq[SubmittedFileAttachment]] = {
+  def publish(files: Seq[FileAttachment])(implicit headerCarrier: HeaderCarrier): Future[Seq[PublishedFileAttachment]] = {
     sequence(
       files.map { f: FileAttachment =>
-        publish(f).recover {
-          case t: Throwable => UnpublishedFileAttachment(f.id, f.name, f.mimeType, f.size, t.getMessage)
+        publish(f).map(Some(_)).recover {
+          case t: Throwable => Logger.error(s"Failed to publish file [${f.id}].", t)
+            None
         }
       }
-    )
+    ).map(_.filter(_.isDefined).map(_.get))
```

Comment:
Seems unnecessary to wrap the files in an `Option` and then call `.get`
There must be a better way to do it. 
One idea I have in my mind right now is to have an extra field of type `Option[Boolean]` with default value `false` in `PublishedFileAttachment`. When the file is published, you update that field.

Reply:
It's because of the recover block

Reply:
I'd stick the recover at the end but then it would fail all the attachments if one of them failed